packages/query-core/src/queryCache.ts

```diff
@@ -178,10 +191,28 @@ export class QueryCache extends Subscribable<QueryCacheListener> {
     return this.queries.find((query) => matchQuery(filters, query))
   }
 
-  findAll(queryKey?: QueryKey, filters?: QueryFilters): Query[]
   findAll(filters?: QueryFilters): Query[]
-  findAll(arg1?: QueryKey | QueryFilters, arg2?: QueryFilters): Query[]
-  findAll(arg1?: QueryKey | QueryFilters, arg2?: QueryFilters): Query[] {
+  /**
+   * @deprecated This method should be used with only one object argument.
+   */
+  findAll(
+    queryKey?: QueryKey,
+    filters?: OmitKeyof<QueryFilters, 'queryKey'>,
+  ): Query[]
+  /**
+   * @deprecated This method should be used with only one object argument.
+   */
+  findAll(
+    arg1?: QueryKey | QueryFilters,
+    arg2?: OmitKeyof<QueryFilters, 'queryKey'>,
+  ): Query[]
+  /**
+   * @deprecated This method should be used with only one object argument.
+   */
+  findAll(
+    arg1?: QueryKey | QueryFilters,
+    arg2?: OmitKeyof<QueryFilters, 'queryKey'>,
+  ): Query[] {
```

Comment:
I tagged jsdoc `@deprecated` on unnecessary type overriding of QueryCache's methods